chia/cmds/param_types.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from decimal import Decimal, InvalidOperation
+from typing import Any, Optional, Union
+
+import click
+
+from chia.cmds.units import units
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.bech32m import bech32_decode, decode_puzzle_hash
+from chia.util.config import selected_network_address_prefix
+from chia.util.ints import uint64
+from chia.wallet.util.address_type import AddressType
+
+
+class TransactionFeeParamType(click.ParamType):
+    """
+    A Click parameter type for transaction fees, which can be specified in XCH or mojos.
+    """
+
+    name: str = "uint64"  # output type
```

Comment:
This might be a nice way to write this so that there's an actual code reference to the type.  Though, depending what the type is and what Click means when they say this is for documentation, other text may be preferred.

```suggestion
    name: str = uint64.__name__  # output type
```

Reply:
i think click intends it to just be "uint64"

Reply:
Isn't that what `uint64.__name__` is?  Except in that case we are referencing a type which makes it more checkable and harder to have a typo like `unt64` or whatever.